documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,301 @@
+  }
+}
+----
+
+The list and add operations are executed for the `STORE_NAME` id. This is a partition key of the data model.
+We can easily retrieve all rows from cassandra using the partition key. They will be sorted by the clustering key.
+The `FruitResource` is using the `FruitService` that encapsulates the data access logic.
+
+When creating the REST API we should not share the same entity object between REST API and data access layers.
+They should not be coupled to allow evolving API independent of the storage layer.
+This is the reason why the API is using the `FruitDto`.
+This class will be used by the quarkus to convert JSON to java object for client request and java object to JSON for the response.
+[source,java]
+----
+public class FruitDto {
+
+  private String name;
+  private String description;
+
+  public FruitDto() {}
+
+  public FruitDto(String name, String description) {
+    this.name = name;
+    this.description = description;
+  }
+  // getters and setters omitted for brevity
+}
+----
+
+One important thing to note is that having a default constructor is required by the JSON serialization layer.
+
+== Configuring the Cassandra database
+The main two properties to configure are `contact-points` to access to Cassandra and `local-datacenter` to use when connecting.
+
+A sample configuration should look like this:
+
+[source,properties]
+----
+# configure the Cassandra client for a replica set of two nodes
+quarkus.cassandra.contact-points={cassandra_ip}:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter={dc_name}
+----
+
+In this example, we are using a single instance running on localhost:
+
+[source,properties]
+----
+# configure the Cassandra client for a single instance on localhost
+quarkus.cassandra.contact-points=127.0.0.1:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter=datacenter1
+----
+
+You can configure all java-driver settings using the `application.conf` or `application.json` files.
+They will be passed automatically to the underlying java-driver config loader.
+The settings from `application.properties` with `quarkus.cassandra` prefix have the priority.
+
+To see the full list of settings, please refer to the {reference-conf-url}[reference.conf]
+
+== Running a Cassandra Database
+As by default, `CassandraClient` is configured to access a local Cassandra database on port 9042 (the default Cassandra port).
+If you have a local running database on this port, you should assert that:
+`quarkus.cassandra.load-balancing-policy.local-datacenter` setting is equal to dc of your local cassandra instance.
+
+If you want to use Docker to run a Cassandra database, you can use the following command to launch one:
+[source,shell]
+----
+docker run \
+   --name local-cassandra-instance \
+   -p 7000:7000 \
+   -p 7001:7001 \
+   -p 7199:7199 \
+   -p 9042:9042 \
+   -p 9160:9160 \
+   -p 9404:9404 \
+   -d \
+   launcher.gcr.io/google/cassandra3
+----
```

Comment:
Along the same lines as what we did in the quickstarter doc we should mention here that 9042 is required, all others are optional (but provide enhanced features like JMX monitoring of the C* instance etc.)